DP-17: Add CLI to Glossary (#5893)
OpenNMS/opennms: docs/modules/reference/pages/glossary.adoc

https://www.discourse.org/[Discourse]:: An open-source forum platform that hosts the https://opennms.discourse.group/[OpenNMS community forums].
There, users can find knowledge base articles, support, announcements, and general discussion about all things OpenNMS.

discovery:: A process that uses ICMP to find network devices and services in your network, then automatically adds them to monitoring (see xref:operation:deep-dive/provisioning/introduction.adoc#discovery-auto[Automatic discovery]).

https://docs.docker.com/[Docker]:: An open-source container virtualization service that lets you deploy application-level services without worrying about operating system maintenance.
{page-component-title} components may optionally be deployed as Docker containers.

Dominion:: The service on a {page-component-title} instance that controls Minion operations (see xref:reference:configuration/minion-confd/minion-confd.adoc[]).

https://www.drools.org/[Drools]:: A system to manage business rules that supports the Java Rules Engine API standard.

> It helps provide a more robust infrastructure for workflow and problem state management in alarmd (see xref:operation:deep-dive/alarms/alarmd.adoc[]).

Author: The xref in the brackets links there. To reduce confusion, I can update the link name to "Alarmd" if you want. :)

events:: Structured historical records of activities in {page-component-title} and the nodes, interfaces, and services it manages (see xref:operation:deep-dive/events/introduction.adoc[]).

== F

flows:: Summaries of network traffic sent by network devices (see xref:operation:deep-dive/flows/introduction.adoc[]).

foreign source:: A unique identifier for a provisioning source that is located outside of your local network.

foreign source definition:: A custom set of service detectors and provisioning policies for a foreign source (see xref:operation:deep-dive/provisioning/foreign-source.adoc[]).

== G

geolocation:: The actual physical location of a node monitored by {page-component-title}, represented by its latitude and longitude.
Nodes can be manually assigned a location, or the location can be automatically determined using the xref:reference:provisioning/adapters/geoip.adoc[GeoIP Provisioning Adapter].

https://grafana.com/[Grafana]:: An open-source analysis and visualization web application that connects to data sources and generates dashboards with charts, graphs, and alerts.

== H

> https://docs.opennms.com/helm/latest[HELM]:: An OpenNMS application that lets you create flexible dashboards to display and customize fault, flow, and performance data from {page-component-title}.

Reviewer: Hot off the presses, we will be updating name from Helm to "OpenNMS plugin for Grafana".

> https://docs.opennms.com/helm/latest[HELM]:: An OpenNMS application that lets you create flexible dashboards to display and customize fault, flow, and performance data from {page-component-title}.

Author: I'll update it to "Grafana plugin" for now, until the larger change in names goes through.

```suggestion
https://docs.opennms.com/helm/latest[HELM]:: An Grafana plugin that lets you create flexible dashboards to display and customize fault, flow, and performance data from {page-component-title}.
```

Originally developed at LinkedIn, it is now supported by the Apache community (see xref:deployment:core/setup-message-broker.adoc[]).

https://www.elastic.co/kibana/[Kibana]:: A component of the Elastic Stack (comprised of Elasticsearch, Logstash, and Kibana).
It enables visualizations for data indexed in an Elasticsearch cluster.

KSC reports:: Key SNMP customized (KSC) reports provide a way to generate prefabricated graphical views of collected data.
They let you display data from different devices and sources (SNMP, ICMP, HTTP) on one page (see xref:operation:deep-dive/visualizations/opsboard/dashlets/ksc.adoc[]).

https://kubernetes.io/[Kubernetes]:: An open-source container orchestration system for automating software deployments, scaling, and management.
Originally designed by Google, the Cloud Native Computing Foundation now maintains it.

== M

https://www.opennms.com/meridian/[Meridian]:: A subscription-based, optimized, and stable version of the OpenNMS Horizon platform that includes only the most stable and secure Horizon features.
It is distributed via annual release with monthly security patches.

> message broker:: When using Minions and Sentinels, a message broker is required for communication between servers.

Reviewer: do we need to define what a message broker is, or can we assume users know that?

Author: ```suggestion
message broker:: An asynchronous service that enables communication between an application's distributed components.
When using Minions and Sentinels, a message broker is required for communication between servers.
```

Minions communicate with these remote devices while {page-component-title} performs coordination and task delegation (see xref:development:minion/introduction.adoc[]).

monitoring location:: Also referred to as "Minion location" or "location."
Nodes assigned to a location are monitored by Minions assigned to the same location. +
This is distinct from geolocation, which is the actual physical location of a monitored entity.

== N

https://github.com/OpenNMS/nephron[Nephron]:: An OpenNMS component that enables horizontal scaling of flow processing.
It is not required for flow processing, but can improve performance for deployments with a very high volume of flow documents.

network operations center (NOC):: One or more locations where a network is monitored and controlled (see https://en.wikipedia.org/wiki/Network_operations_center[Network Operations Center]).

xref:deployment:time-series-storage/newts/introduction.adoc[New-Fangled Time Series Datastore (Newts)]:: An OpenNMS time series datastore based on Apache Cassandra.
You can use it as an alternative persistence strategy (instead of JRobin or RRDtool).

> node:: A device or application that sends communications over a network (for example, a server).

Reviewer: can nodes also include interfaces?

> They can be provisioned to and monitored by {page-component-title}.

Author: ```suggestion
They can be provisioned to and monitored by {page-component-title}.
A node must include at least one IP interface, otherwise it is removed from your {page-component-title} inventory.
```

Nodes assigned to a location are monitored by Minions assigned to the same location. +
This is distinct from geolocation, which is the actual physical location of a monitored entity.

== N

https://github.com/OpenNMS/nephron[Nephron]:: An OpenNMS component that enables horizontal scaling of flow processing.
It is not required for flow processing, but can improve performance for deployments with a very high volume of flow documents.

network operations center (NOC):: One or more locations where a network is monitored and controlled (see https://en.wikipedia.org/wiki/Network_operations_center[Network Operations Center]).

xref:deployment:time-series-storage/newts/introduction.adoc[New-Fangled Time Series Datastore (Newts)]:: An OpenNMS time series datastore based on Apache Cassandra.
You can use it as an alternative persistence strategy (instead of JRobin or RRDtool).

node:: A device or application that sends communications over a network (for example, a server).
They can be provisioned to and monitored by {page-component-title}.
A node must include at least one IP interface, otherwise it is removed from your {page-component-title} inventory.

> node availability:: The percentage of time that a node is detected as being accessible.

Reviewer 2: This may just be "availability" under the "A" section

Author: I know that we split "availability" into multiple definitions in the internal Glossary that we were working on - would it make more sense to simply keep them all under one definition? Are they distinct enough to require multiple definitions?
